core/dyn-configs/src/lib.rs

```diff
@@ -1,28 +1,60 @@
 #![doc = include_str!("../README.md")]
 
-use near_o11y::metrics::{try_create_int_counter, IntCounter};
-use once_cell::sync::Lazy;
-use std::sync::atomic::{AtomicU64, Ordering};
-
-/// An indicator for dynamic config changes
-pub static DYN_CONFIG_CHANGE: Lazy<IntCounter> = Lazy::new(|| {
-    try_create_int_counter(
-        "near_dynamic_config_changes",
-        "Total number of dynamic configuration changes",
-    )
-    .unwrap()
-});
-
-// NOTE: AtomicU64 is the same unit as BlockHeight, and use to store the expected blockheight to
-// shutdown
-pub static EXPECTED_SHUTDOWN_AT: AtomicU64 = AtomicU64::new(0);
-
-/// Reload the dynamic config, and increase the counting metric near_dynamic_config_changes
-pub fn reload(expected_shutdown: Option<u64>) {
-    if let Some(expected_shutdown) = expected_shutdown {
-        EXPECTED_SHUTDOWN_AT.store(expected_shutdown, Ordering::Relaxed);
-    } else {
-        EXPECTED_SHUTDOWN_AT.store(0, Ordering::Relaxed);
+use near_chain_configs::Consensus;
+use near_o11y::log_config::LogConfig;
+use serde::{Deserialize, Serialize};
+use tokio::sync::broadcast::Sender;
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct DynConfig {
+    /// Graceful shutdown at expected blockheight.
+    pub expected_shutdown: Option<u64>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Default)]
+pub struct DynConfigs {
+    pub dyn_config: Option<DynConfig>,
+    pub log_config: Option<LogConfig>,
+    pub consensus: Option<Consensus>,
+}
+
+#[derive(Default)]
+pub struct DynConfigStore {
+    dyn_configs: DynConfigs,
+    original_consensus: Consensus,
+    tx: Option<Sender<DynConfigs>>,
+}
+
+impl DynConfigStore {
+    pub fn reload(&mut self, mut dyn_configs: DynConfigs) {
+        if dyn_configs.consensus.is_none() {
+            dyn_configs.consensus = Some(self.original_consensus.clone());
```

Comment:
To confirm, this logic is for that if the user deletes the consensus file in the dynamic folder, consensus config will be reset to the original values, is that right? If so, please add comments in `DynConfigs`. It is not obvious that this is the intended behavior.

Reply:
There is no consensus file. Consensus is read directly from `config.json`.
Added some comments at the definition of those structs.